src/common/mem.c

```diff
@@ -38,23 +38,34 @@ enum {
 	DAOS_MD_PMEM	= 0,
 	DAOS_MD_BMEM	= 1,
 	DAOS_MD_ADMEM	= 2,
+	DAOS_MD_INVAL	= 4,
 };
 
 static int daos_md_backend = DAOS_MD_PMEM;
 
-/** Sets up global settings for the pmem objects.
+/** Initializes global settings for the pmem objects.
+ *
+ *  \param	md_on_ssd[IN]	Boolean indicating if MD-on-SSD is enabled.
  *
  *  \return	0 on success, non-zero on failure.
  */
 int
-umempobj_settings_init(void)
+umempobj_settings_init(bool md_on_ssd)
 {
 	int					rc;
 	enum pobj_arenas_assignment_type	atype;
-	unsigned int				val = DAOS_MD_PMEM;
+	unsigned int				md_mode = DAOS_MD_INVAL;
+
+	d_getenv_int("DAOS_MD_ON_SSD", &md_mode);
+
+	if (md_on_ssd && md_mode == DAOS_MD_PMEM) {
+		D_ERROR("MD on SSD mode detected but DAOS_MD_ON_SSD set to MD_PMEM (0)");
+	}
```

Comment:
Now we get "md-on-ssd enabled or not" from config file, and the "DAOS_MD_ON_SSD" env var is only used to specify "ADMEM" or "BMEM" when "md-on-ssd enabled", so this won't never happen.

Reply:
I haven't changed the behaviour of the environment variable, so the idea is that it can still be used to override the decision from the config file (for cases like unit tests). This check prints warning if we override with MD_PMEM when we have detected MD-on-SSD in the config.

Reply:
in unit tests for example the md_on_ssd will always be false, so we need a way to synthesise it

Reply:
suggested change made. Done